### PR TITLE
Allow services in same thrift to have same method name

### PR DESCRIPTION
**Pull request by jyang0**
- in generated code, added service name prefix to method names to tell them apart
- added a new example thrift with 2 services having same method name
- updated existing test data

### CODE REVIEW

File: backend/repository/data/handler/test_cases.json [hunk 1]
```diff
@@ -2700,6 +2764,18 @@
 							"thriftFile": "clients/googlenow/googlenow.thrift",
 							"type": "http"
 						},
+						"multi": {
+							"exposedMethods": {
+								"HelloA": "ServiceABack::hello",
```
Comment (cl1337) [inline backend/repository/data/handler/test_cases.json]: we'll have to fix how `HelloA` is generated in here: https://github.com/uber/zanzibar/blob/master/backend/repository/client_update.go#L89

I believe now it still use method name as key, that will be overriden

File: codegen/method.go [hunk 2]
```diff
@@ -881,7 +881,7 @@ func (ms *MethodSpec) setTypeConverters(
 
 	respConverter.append(
 		"func convert",
-		pascalCase(ms.Name),
+		ms.DownstreamService, pascalCase(ms.Name),
```
Comment (ChuntaoLu) [inline codegen/method.go]: While conventionally service name is in pascal case, but it's not enforced. So let's apply `pascalCase` on the service name as well.

File: codegen/module_system.go [hunk 3]
```diff
@@ -839,7 +839,7 @@ func (g *EndpointGenerator) generateEndpointFile(
 
 	var workflowName string
 	if method.Downstream != nil {
-		workflowName = strings.Title(method.Name) + "Endpoint"
+		workflowName = thriftServiceName + strings.Title(method.Name) + "Endpoint"
```
Comment (ChuntaoLu) [inline codegen/module_system.go]: Similarly here, let's `string.Title` the service name.

File: codegen/method.go (hunk 2 above)
Comment (jyang0) [reply codegen/method.go]: Got it!

**Pull request merged**